fetching/directory-fetcher/src/index.ts

```diff
@@ -83,14 +86,16 @@ async function _fetchAllFilesFromDir (
       const { filePath, stat } = fileStatResult
       const relativeSubdir = `${relativeDir}${relativeDir ? '/' : ''}${file}`
       if (stat.isDirectory()) {
-        const subFilesIndex = await _fetchAllFilesFromDir(readFileStat, filePath, relativeSubdir)
-        Object.assign(filesIndex, subFilesIndex)
+        const subFetchResult = await _fetchAllFilesFromDir(readFileStat, filePath, relativeSubdir)
+        Object.assign(filesIndex, subFetchResult.filesIndex)
+        Object.assign(filesStats, subFetchResult.filesStats)
       } else {
         filesIndex[relativeSubdir] = filePath
+        filesStats[relativeSubdir] = fileStatResult.stat
       }
     })
   )
-  return filesIndex
+  return { filesIndex, filesStats }
```

Comment:
This bothers me a bit because we only need stats in update-injected-packages but now we'll return it also for all the installed directory dependencies. Which might be not a big deal as directory dependencies/injected dependencies are rare.

Reply:
The cost of this feature would be a few more object assignments and a bit more temporary objects for the GC to clean up. Other than that, there is no additional syscalls. The APIs aren't broken either because `_fetchAllFilesFromDir` (the function whose return type changes in a major way) is a private/internal function. The gain of this feature is that there would be fewer calls to `fs.stat` when calling `DirFetcher.apply()`.